### PR TITLE
Compression algorithm changes

### PR DESCRIPTION
**Pull request by BanzaiMan**
* When downloading, download to `fetch.tgz` or `fetch.tbz` based on the file we're downloading.
* When uploading, use the correct compression algorithm based on the file name.

We will assume that `travis-build` will give us the list of URLs to download the cache from in the desired order.

### CODE REVIEW

File: bin/casher [hunk 1]
```diff
@@ -88,10 +88,16 @@ class Casher
...
-    command = "tar -Pj#{flag}f #{Shellwords.escape(file)} #{Shellwords.join(args)}"
+    compression_flag = file.end_with?('.tbz') ? 'j' : 'z'
+    command = "tar -P#{compression_flag}#{flag}f #{Shellwords.escape(file)} #{Shellwords.join(args)}"
     block ||= proc { puts "FAILED: #{command}", File.read("#{@casher_dir}/tar.err.log"), File.read("#{@casher_dir}/tar.log") }
     block.call unless system "#{command} 2>#{@casher_dir}/tar.err.log >#{@casher_dir}/tar.log"
   end
+
+  private
+  def download_from(urls)
+    (urls.map {|url| url.gsub(/\.tbz(\?)?\b/, '.tgz\1')} + urls).uniq
```
Comment (BanzaiMan) [inline bin/casher]: Actually, there is a chance that this is insufficient; changing the file's extension may change the required signature for fetching.

If this is indeed the case, URL's must be constructed on the `travis-build` side.

Comment (BanzaiMan) [inline bin/casher]: Staging doesn't have caching enabled, so I can't test this hypothesis until travis-ci/travis-build#468 is deployed.

**Comment by BanzaiMan** [comment]
I've tested@8acab93, and confirmed that it uploads the archive compressed with bzip2.

**Comment by joshk** [comment]
awesome!

**Pull request merged**